src/main/java/org/commcare/formplayer/application/WebAppContext.java

```diff
@@ -323,4 +324,8 @@ public LockProvider lockProvider(DataSource dataSource) {
                         .build()
         );
     }
+
+    @Bean public RequestContextListener requestContextListener(){
```

Comment:
this is the recommended way to make the request object available.

The only changes to non-test code are related to this - accessing the request from the `RequestContextHolder`